src/redis-cli.c

```diff
@@ -1820,6 +1829,11 @@ static int parseOptions(int argc, char **argv) {
         exit(1);
     }
 
+    if (config.output != OUTPUT_JSON && config.resp2 && config.resp3) {
```

Comment:
this will also look cleaner if we consider a config.resp3 value of 2 and implicit one.
i.e. we only fail if the explicit configs given by the user are conflicting.